api/controllers/v1/slack.ts

```diff
@@ -100,58 +100,35 @@ const deployHelper = (deployable, payload, jobTitle, jobUserName, jobUserEmail)
-  if (!slackConnector.validateSlackRequest(event)) {
-    return prepReponse(401, 'text/plain', 'Signature Mismatch, Authentication Failed!');
-  }
-  const client = new mongodb.MongoClient(c.get('dbUrl'));
-  await client.connect();
-  const db = client.db(c.get('dbName'));
-  const repoEntitlementRepository = new RepoEntitlementsRepository(db, c, consoleLogger);
-  const branchRepository = new BranchRepository(db, c, consoleLogger);
-  const jobRepository = new JobRepository(db, c, consoleLogger);
-  const parallel = c.get<any>('parallel');
-  const env = c.get<string>('env');
+// For every repo/branch selected to be deployed, return an array of jobs with the payload data
+// needed for a successful build.
+export const getDeployableJobs = async (values, entitlement, branchRepository: BranchRepository) => {
   const deployable = [];
-  // This is coming in as urlencoded stirng, need to decode before parsing=
-
-  const decoded = decodeURIComponent(event.body).split('=')[1];
-  const parsed = JSON.parse(decoded);
-  const stateValues = parsed.view.state.values;
 
-  const entitlement = await repoEntitlementRepository.getRepoEntitlementsBySlackUserId(parsed.user.id);
-  if (!isUserEntitled(entitlement)) {
-    return prepReponse(401, 'text/plain', 'User is not entitled!');
-  }
-
-  const values = slackConnector.parseSelection(stateValues);
-  let jobCount = 0;
   for (let i = 0; i < values.repo_option.length; i++) {
-    // // e.g. mongodb/docs-realm/master => (site/repo/branch)
+    // e.g. mongodb/docs-realm/master => (site/repo/branch)
     const [repoOwner, repoName, branchName] = values.repo_option[i].value.split('/');
     const hashOption = values?.hash_option ?? null;
     const jobTitle = `Slack deploy: ${values.repo_option[i].value}, by ${entitlement.github_username}`;
     const jobUserName = entitlement.github_username;
     const jobUserEmail = entitlement?.email ?? '';
 
     const repoInfo = await branchRepository.getRepo(repoName);
-    const non_versioned = repoInfo.branches.length === 1 ? true : false;
+    const non_versioned = repoInfo.branches.length === 1;
 
     const branchObject = await branchRepository.getRepoBranchAliases(repoName, branchName);
     if (!branchObject?.aliasObject) continue;
 
     const publishOriginalBranchName = branchObject.aliasObject.publishOriginalBranchName; //bool
-    let aliases = branchObject.aliasObject.urlAliases; //array or null
-    let urlSlug = branchObject.aliasObject.urlSlug; //string or null, string must match value in urlAliases or gitBranchName
+    let aliases = branchObject.aliasObject.urlAliases; // array or null
+    let urlSlug = branchObject.aliasObject.urlSlug; // string or null, string must match value in urlAliases or gitBranchName
     const isStableBranch = branchObject.aliasObject.isStableBranch; // bool or Falsey
     aliases = aliases?.filter((a) => a);
-    if (!urlSlug || !!urlSlug.trim()) {
+    if (!urlSlug || !urlSlug.trim()) {
```

Comment:
For context, having `!!urlSlug.trim()` would cause this if statement to always evaluate as true. This would cause the `urlSlug` to always be set to the current branch's name, causing there to be a mismatch when assigning `primaryAlias`.